<a href="https://colab.research.google.com/github/Anacezegarra/programa-python/blob/main/Ana_Cecilia_Zegarra.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Crea una lista llamada numeros_pares que contenga los números del 1 al 100 con saltos de 2 en 2

In [4]:
numeros_pares = list(range(2, 101, 2))
print(numeros_pares)
type(numeros_pares)

[2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100]


list

Agrega los números 102, 104 y 106 al final de la lista numeros_pares y luego
elimina los primeros 5 números de la lista numeros_pares. Muestra la lista
resultante

In [5]:
numeros_pares.append(102)
numeros_pares.append(104)
numeros_pares.append(106)
print(numeros_pares)

[2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106]


In [6]:
del numeros_pares[:5]
print(numeros_pares)

[12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106]


Crea una lista llamada resultante que contenga la raíz cuadrada de cada
elemento de la lista numeros_pares usando list comprehension.

In [7]:
import math
resultante = [math.sqrt(numero) for numero in numeros_pares]
print(resultante)

[3.4641016151377544, 3.7416573867739413, 4.0, 4.242640687119285, 4.47213595499958, 4.69041575982343, 4.898979485566356, 5.0990195135927845, 5.291502622129181, 5.477225575051661, 5.656854249492381, 5.830951894845301, 6.0, 6.164414002968976, 6.324555320336759, 6.48074069840786, 6.6332495807108, 6.782329983125268, 6.928203230275509, 7.0710678118654755, 7.211102550927978, 7.3484692283495345, 7.483314773547883, 7.615773105863909, 7.745966692414834, 7.874007874011811, 8.0, 8.12403840463596, 8.246211251235321, 8.366600265340756, 8.48528137423857, 8.602325267042627, 8.717797887081348, 8.831760866327848, 8.94427190999916, 9.055385138137417, 9.16515138991168, 9.273618495495704, 9.38083151964686, 9.486832980505138, 9.591663046625438, 9.695359714832659, 9.797958971132712, 9.899494936611665, 10.0, 10.099504938362077, 10.198039027185569, 10.295630140987]


Dada la siguiente tupla tupla_x = (“alumno”,”Pedro”,15), desempaqueta la
tupla tupla_x en tres variables: x, y, z, e imprime la siguiente estructura “El
alumno Pedro obtuvo la calificación de 15 usando las variables x, y, z”.

In [8]:
tupla_x = ("alumno", "Pedro", 15)
x, y, z = tupla_x
print(f"El {x} {y}  obtuvo la calificación de {z}")

El alumno Pedro  obtuvo la calificación de 15


Dado el siguiente diccionario notas = {“Programacion”:15.9,
“Matematica”: 15.1, “Fisica”:15.6}, escribe un bucle for que imprima
cada clave y su valor en el diccionario con la siguiente estructura “Mis
calificaciones en el curso de Programacion es 15.9”

In [9]:
notas = {
    "Programacion": 15.9,
    "Matematica": 15.1,
    "Fisica": 15.6
}
for curso, nota in notas.items():
  print(f"Mis calificaciones en el curso de {curso} es {nota}")

Mis calificaciones en el curso de Programacion es 15.9
Mis calificaciones en el curso de Matematica es 15.1
Mis calificaciones en el curso de Fisica es 15.6


Crea una matriz 1D que contenga los primeros 16 elementos de la lista
numeros_pares (creada en la pregunta 1), modifique las dimensiones y
convertiela en una matriz 2D con shape(4,4), calcula la suma total de los
elementos de la matriz matriz 2D.

In [10]:
primeros_16 = numeros_pares[:16]
print(primeros_16)

[12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42]


In [11]:
import numpy as np
matriz_1d = np.array(primeros_16)
print(matriz_1d)

[12 14 16 18 20 22 24 26 28 30 32 34 36 38 40 42]


In [12]:
matriz_2d = matriz_1d.reshape(4,4)
print(matriz_2d)

[[12 14 16 18]
 [20 22 24 26]
 [28 30 32 34]
 [36 38 40 42]]


In [13]:
suma_matriz = matriz_2d.sum()
print(suma_matriz)

432


Leer el archivo datosDemograficos.xlsx y realice un análisis descriptivo en
función de las variables más relevantes.


In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [15]:
d = pd.read_excel('/content/datosDemograficos (1).xlsx')

In [16]:
d.head()

Indicadores  Fecundidad  Mortalidad  TasaMortalidadxMil  EsperanzaVida  \
0        Puno       30108        9951                6.86          71.44   
1  San Martín       16074        4889                5.63          72.45   
2       Tacna        5605        1900                5.40          75.11   
3      Tumbes        3924        1209                4.94          74.96   
4     Ucayali        8872        3020                5.93          71.99   

   EVSoloHombres  EVSoloMujeres  TasaMortalidadInfantilxMilNacidos  \
0          68.77          74.25                              26.82   
1          69.65          75.40                              17.53   
2          72.31          78.06                              11.77   
3          72.09          77.98                              11.46   
4          69.24          74.87                              21.07   

   CrecimientoNatural  
0               20157  
1               11185  
2                3705  
3                2715  
4                5852

In [18]:
print(d.describe())

          Fecundidad   Mortalidad  TasaMortalidadxMil  EsperanzaVida  \
count      25.000000     25.00000           24.000000      25.000000   
mean    22874.400000   7265.28000            5.711667      74.059600   
std     30817.223005  10451.42713            0.580777       2.375038   
min      2571.000000    660.00000            4.540000      70.930000   
25%      8872.000000   2788.00000            5.397500      71.990000   
50%     16074.000000   5234.00000            5.545000      73.290000   
75%     24385.000000   8450.00000            5.985000      76.450000   
max    163089.000000  55201.00000            6.970000      78.380000   

       EVSoloHombres  EVSoloMujeres  TasaMortalidadInfantilxMilNacidos  \
count      25.000000      25.000000                          25.000000   
mean       71.337200      76.919200                          16.984000   
std         2.315498       2.443766                           5.434434   
min        68.530000      73.450000                    

**Fecundidad:** Promedio de 22,874 nacimientos, pero con una gran dispersión (std ≈ 30,817), hay territorios con tasas muy distintas entre sí. El rango va de 2,571 a 163,089.

**Mortalidad:**(Promedio ≈ 7,265, std ≈ 10,451), con un máximo muy alto (55,201), con demasiada variabilidad lo que podría indicar la presencia de un outlier que represente una región con condiciones críticas.

**General:** Media de 74.05 años, con un mínimo de 70.93 y máximo de 78.38. La dispersión es baja, lo que sugiere estabilidad general en las condiciones de vida.

Hombres: Esperanza media ≈ 71.34 años.
Mujeres: Esperanza media más alta, ≈ 76.92 años.

Las mujeres viven, en promedio, 5.58 años más que los hombres.

**Mortalidad nacidos:** Promedio de 16.98 muertes por mil nacidos, con un amplio rango de 7.81 a 26.82. Esto puede reflejar desigualdades importantes en acceso a servicios de salud para recién nacidos.

**Crecimiento Natural:** Promedio 15,609 personas, pero con una gran desviación estándar (std ≈ 20,410), lo cual refleja un comportamiento muy desigual.

El valor máximo (107,888) es muy alto comparado con el mínimo (1,566), lo que sugiere la existencia de una o varias regiones con un crecimiento demográfico exponencial.